### DISCUSSION TITLE
Unable to connect PostgreSQL sever

### DISCUSSION THREAD

**Discussion by MrShakila** [serverpod/serverpod#856] [Q&A] [answered]
I am getting this error when i try to deploy the sever pod sever 

`Failed to connect to the database. Retrying in 10 seconds. PostgreSQLSeverity.fatal 28P01: password authentication failed for user "postgres" `

![Screenshot 2023-04-28 at 17 04 51](https://user-images.githubusercontent.com/90470539/235137219-34583d2a-220f-45cd-b010-f75b70a2497c.png)

Flutter Doctor Results

![Screenshot 2023-04-28 at 17 06 44](https://user-images.githubusercontent.com/90470539/235137557-dd8415e1-f271-486c-b249-e04e44e339a1.png)

**Comment by Isakdl** [comment]
You are not running another instance of postgres on the same port?

If you run `docker ps` what do you see?

**Comment by avatarnguyen** [comment]
I had a similiar case where my password is not correct. maybe double check your `passwords.yaml` credential with your `development.yaml` passwords for the database

**Comment by yahu1031** [comment]
@MrShakila Duplicate #555

**Reply by MarkOSullivan94** [reply]
I've also got the same issue and none of the solutions are appropriate when you are already making use of Docker containers for other apps

**Reply by yahu1031** [reply]
Correct. But volumes in the containers are the issues. Make sure you are handling them correctly.

**Comment by Isakdl** [comment]
Perhaps you recreated the project with the same name? Like others have suggested this is due to some old volumes that was not deleted.

If you run
`docker compose down -v` from the server project, this will remove all containers AND the volumes! Then when you rerun `docker compose up --build --detach` the volume is recreated with the correct data. This way you do not have to touch any other docker configurations either! @MarkOSullivan94

An FYI the volume should be called something like `project_server_project_data` where project is your project name. You can list them by running `docker volume ls` and `docker volume rm <name>` to delete them.

**Reply by MarkOSullivan94** [reply]
> docker compose down -v from the server project, this will remove all containers AND the volumes! Then when you rerun docker compose up --build --detach the volume is recreated with the correct data. This way you do not have to touch any other docker configurations either! @MarkOSullivan94

Wasn't aware of this! I'll try this out next time I'm working on my serverpod project. Thank you for sharing!

**Reply by MrShakila** [reply]
I don't know if this answer fixed my issue, but when you said there is can be another container, then I deleted my all containers and reinstalled severpod then it works, thanks mate

**Reply by MrShakila** [reply]
well I tried that and it's not working, the problem is my server is up and running before,but today it is not working
i am still getting this error

```
SERVERPOD version: 1.1.0, dart: 2.19.6 (stable) (Tue Mar 28 13:41:04 2023 +0000) on "macos_x64", time: 2023-05-17 05:35:22.040345Z
mode: development, role: monolith, logging: normal, serverId: default
Failed to connect to the database. Retrying in 10 seconds. PostgreSQLSeverity.error 42P01: relation "serverpod_runtime_settings" does not exist 
Database configuration:
database host: localhost
database port: 8090
database name: findmypet
database user: postgres
database require SSL: false
database unix socket: false
database pass: ********

Failed to connect to the database. Retrying in 10 seconds. PostgreSQLSeverity.error 42P01: relation "serverpod_runtime_settings" does not exist 
Failed to connect to the database. Retrying in 10 seconds. PostgreSQLSeverity.error 42P01: relation "serverpod_runtime_settings" does not exist 
Failed to connect to the database. Retrying in 10 seconds. PostgreSQLSeverity.error 42P01: relation "serverpod_runtime_settings" does not exist 
Failed to connect to the database. Retrying in 10 seconds. PostgreSQLSeverity.error 42P01: relation "serverpod_runtime_settings" does not exist 
```

**Reply by yahu1031** [reply]
@MrShakila Duplicate of #893

**Reply by MrShakila** [reply]
@yahu1031 that doesn't resolve my issue

**Reply by yahu1031** [reply]
Did you create tables in DB? @MrShakila

**Reply by MrShakila** [reply]
no didn't change the serverside yet

**Reply by Isakdl** [reply, marked as answer]
If you kill the container, you also delete all database tables, you will have to recreate them again by running the SQL code inside the `your_project_server/generate` folder.

When running `serverpod create my_project` the cli will create all the tables for you during the init process (unless you are on windows). In this case you deleted the containers afterwards hence you need to recreate them.

There are plenty of ways to do that, such as just executing the code in a db viewer, or you can use docker to execute the SQL code directly in the database. 

Take a look at this https://stackoverflow.com/questions/76180598/serverpod-sql-error-when-starting-a-clean-project for the commands to run on docker.

**Reply by MrShakila** [reply]
yes ,you are correct i just need to connect PostgreSQL DB and create the tables then it works perfectly
